src/components/Editor.tsx

```diff
@@ -441,6 +474,21 @@ export const Editor = ({
     getThemeLocalStorageKey,
   ]);
 
+  // Handles sending the sendDevThemeSaveStateData on initial load so that nothing isn't rendered for preview.
+  // Subsequent changes are sent via handleChange in ThemeSidebar.tsx.
+  useEffect(() => {
+    if (!themeSaveStateFetched || !templateMetadata?.isThemeMode) {
+      return;
+    }
+
+    const themeToSend = themeSaveState?.history[themeSaveState.index];
+
+    devLogger.logFunc("sendDevThemeSaveStateData useEffect");
+    sendDevThemeSaveStateData({
+      payload: { devThemeSaveStateData: JSON.stringify(themeToSend) },
+    });
+  }, [themeSaveStateFetched, themeSaveState, templateMetadata]);
```

Comment:
Do you want themeSaveStateFetched here instead of themeInitialHistoryFetched? 

Above you are using puckInitialHistoryFetched.

Reply:
Good catch